Use stela endpoints for `getRecord`

slifty opened the pull request:
This PR updates our `getRecord` call to use stela. This improvement is necessary to support unlisted share functionality, which only exists on stela.

This PR represents a modest refactor from the one opened at #630; the primary difference is that we re-package v2 results so that the object returned is the same type as before (e.g. a `RecordResponse` vs a `RecordVO`).  This should prevent the need to refactor our components to have v1 and v2 versions as part of the unlisted shares improvements.

Related to #683

cecilia-donnelly reviewed (dismissed):
Very exciting! I was able to share a file and create a share link, and the site worked as expected, so it looks like this is working. Google Maps didn't load locally, but I suspect that's a local environment problem.

slifty commented:
@cecilia-donnelly I'm doing some local QA and saw keywords acting funny -- going to go through with a fine-toothed-comb to see what similar "mocking" needs to be done to turn StelaRecords to RecordVos but overall the approach is still gonna be great

slifty commented:
@cecilia-donnelly ready for a re-review!

slifty commented:
It looks like tests may be *actually* failing!  Taking a look now.

slifty commented:
Fixed the tests (I had to update the test fixture response to reflect stela return format).

I also fixed some leaking tests while I was in the area, and added that as another commit.

cecilia-donnelly reviewed (changes requested):
We talked about this a little bit elsewhere, but to recap here:
- Tags are not appearing specifically in the side/detail view on the file list (what you see when you single click an item in a workspace)

The other problems I saw resolved, so were likely related to queuing. I focused on QA-ing folder and record actions, so only lightly touched account, archive, and billing related aspects of the app. It's always possible that those are related in arcane ways, but seems less likely.

Here is my QA notes doc (sorry, not public): https://docs.google.com/document/d/1khJX_VNwCcES9dGHXDRwIsUXcnW06_uIuXi7L6oTrRM/edit?tab=t.0

slifty commented on src/app/shared/services/api/record.repo.ts:
```diff
@@ -37,23 +47,203 @@ class MultipartUploadUrlsList {
+interface StelaFile {
+	size: number;
+	type: string;
+	fileId: number;
```
Ty!  Replying here but noted and fixing.

slifty commented:
@slifty given that you gave this a pretty deep QA do you think could count as QA from a process PoV?

slifty commented:
Fixed up (the issue was that my Stela type didn't account for the fact that `tags` can be `null`.)

Time for a new round of QA!  Going through the document.
